packages-exp/auth-exp/src/core/strategies/custom_token.ts

```diff
@@ -56,3 +63,54 @@ export async function signInWithCustomToken(
   await authInternal._updateCurrentUser(cred.user);
   return cred;
 }
+
+/**
+ * Sets a custom token provider.
+ *
+ * @remarks
+ * The custom token provider is invoked when a new Firebase ID token is requested and no
+ * refresh token is present (i.e., in passthrough mode). This callback should be implemented to
+ * obtain a custom token to exchange for a new Firebase ID token. For instance,
+ *
+ * ```js
+ * setCustomTokenProvider(auth, {
+ *   async getCustomToken(): Promise<string> {
+ *     return requestNewCustomToken();
+ *   }
+ * });
+ * ```
+ *
+ * @param auth - The Auth instance.
+ * @param provider - The callback that is invoked when a new Firebase ID token is requested and no
+ * refresh token is present.
+ */
+export function setCustomTokenProvider(
+  auth: Auth,
+  provider: CustomTokenProvider
+): void {
+  const authInternal = _castAuth(auth);
```

Comment:
Nice, this looks great and is fully tree-shakeable. @rosalyntan this is the new implementation we chatted about yesterday

Reply:
Please call `getModularInstance` on the auth variable first, then call `_castAuth` with the result. For example:
https://github.com/firebase/firebase-js-sdk/blob/2901ed05e946d7010443fd3fa48b4aa3395e5463/packages-exp/auth-exp/src/core/index.ts#L57-L62

Users can pass the compat layer auth instance to the modular functions (for migrations) so the first step is to unwrap if necessary